File: frontend/src/types/agents.ts
import type { Severity, RiskLevel, ServiceStatus, ImpactLevel } from './domain'

export interface TriageResult {
  severity: Severity
  category: string
  tags: string[]
  is_duplicate: boolean
  summary: string
}

export interface LogFindings {
  error_pattern: string
  first_occurrence: string
  frequency: string
  affected_versions: string[]
  last_healthy_version: string
  correlated_event: string | null
  sample_stack_trace: string | null
}

export interface AffectedService {
  service: string
  status: ServiceStatus
  impact: ImpactLevel
}

export interface InvestigationResult {
  log_findings: LogFindings
  affected_services: AffectedService[]
  estimated_users_affected: string
}

export interface RootCauseResult {
  probable_cause: string
  confidence_score: number
  evidence: string[]
  contributing_factors: string[]
}

export interface RemediationOption {
  id: string
  title: string
  description: string
  confidence: number
  risk_level: RiskLevel
  estimated_recovery_time: string
  steps: string[]
}

export interface PRInfo {
  pr_number: number
  pr_url: string
  diff: string
  file_path: string
  branch: string
}

export interface RemediationResult {
  options: RemediationOption[]
  pr?: PRInfo
}

export interface DemoScenario {
  id: string
  title: string
  service: string
  severity: string
  language: string
  description: string
}

export interface HumanDecision {
  approved_option_id: string
  approved_by: string
  notes: string | null
}

export interface PostMortemImpact {
  users_affected: string
  services_degraded: string[]
}

export interface PostMortemTimelineEntry {
  time: string
  event: string
}

export interface LessonsLearned {
  went_well: string[]
  went_wrong: string[]
  got_lucky: string[]
}

export interface PostMortem {
  title: string
  duration: string
  severity: Severity
  summary: string
  impact: PostMortemImpact
  timeline: PostMortemTimelineEntry[]
  root_cause: string
  contributing_factors: string[]
  remediation_taken: string
  lessons_learned: LessonsLearned
  prevention: string[]
}

export interface SimilarIncident {
  incident_id: string
  service: string
  severity: Severity
  summary: string
  root_cause: string
  remediation_taken: string
  resolved_at: string
  similarity_score: number
}

File: api/app/schemas/agents.py
from typing import Literal

from pydantic import BaseModel, Field


class LogFindings(BaseModel):
    error_pattern: str
    first_occurrence: str
    frequency: str
    affected_versions: list[str]
    last_healthy_version: str
    correlated_event: str | None
    sample_stack_trace: str | None


class AffectedService(BaseModel):
    service: str
    status: Literal["down", "degraded", "healthy"]
    impact: Literal["primary", "downstream", "none"]


class RemediationOption(BaseModel):
    id: str
    title: str
    description: str
    confidence: float = Field(ge=0.0, le=1.0)
    risk_level: str
    estimated_recovery_time: str
    steps: list[str]


class PostMortemImpact(BaseModel):
    users_affected: str
    estimated_revenue_loss: str | None
    services_degraded: list[str]


class PostMortemTimelineEntry(BaseModel):
    time: str
    event: str


class TriageResult(BaseModel):
    severity: str
    category: str
    tags: list[str]
    is_duplicate: bool
    summary: str


class InvestigationResult(BaseModel):
    log_findings: LogFindings
    affected_services: list[AffectedService]
    estimated_users_affected: str
    revenue_impact_per_minute: str | None


class RootCauseResult(BaseModel):
    probable_cause: str
    confidence_score: float = Field(ge=0.0, le=1.0)
    evidence: list[str]
    contributing_factors: list[str]


class PRInfo(BaseModel):
    pr_number: int
    pr_url: str
    diff: str
    file_path: str
    branch: str


class RemediationResult(BaseModel):
    options: list[RemediationOption]
    pr: PRInfo | None = None


class HumanDecision(BaseModel):
    approved_option_id: str
    approved_by: str
    notes: str | None


class PostMortem(BaseModel):
    title: str
    duration: str
    severity: str
    impact: PostMortemImpact
    timeline: list[PostMortemTimelineEntry]
    root_cause: str
    remediation_taken: str
    prevention: list[str]
